webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx

```diff
@@ -75,6 +75,34 @@ const ExperimentSingleTrialTabs: React.FC<Props> = (
     }
   }, [ basePath, history, tab, tabKey ]);
 
+  const fetchFirstTrialId = useCallback(async () => {
+    try {
+      const expTrials = await getExpTrials(
+        { id: experiment.id, limit: 2 },
+        { signal: canceler.signal },
+      );
+      if (expTrials.trials[0]) {
+        setFirstTrialId(expTrials.trials[0].id);
+      }
+    } catch (e) {
+      handleError({
+        error: e,
+        level: ErrorLevel.Error,
+        message: e.message,
+        publicMessage: 'Failed to fetch experiment trials.',
+        silent: true,
+        type: ErrorType.Server,
+      });
+    }
+  }, [ canceler, experiment.id ]);
+
+  const { stopPolling: stopPollingFirstTrialId } = usePolling(fetchFirstTrialId);
+
+  useEffect(() => {
+    if (trialId != null) return;
+    return () => stopPollingFirstTrialId();
+  }, [ trialId, stopPollingFirstTrialId ]);
+
```

Comment:
**nit:** move polling next to the polling setup code below and same with the useEffect?